processor/src/test/java/org/mapstruct/ap/testutil/runner/CompilingStatement.java

```diff
@@ -110,32 +118,48 @@ protected void setupDirectories() throws Exception {
 
         classOutputDir = compilationRoot + "/classes";
         sourceOutputDir = compilationRoot + "/generated-sources";
+        additionalCompilerClasspath = compilationRoot + "/compiler";
 
         createOutputDirs();
 
-        ( (ModifiableURLClassLoader) Thread.currentThread().getContextClassLoader() ).addOutputDir( classOutputDir );
+        ( (ModifiableURLClassLoader) Thread.currentThread().getContextClassLoader() ).withPath( classOutputDir );
     }
 
     protected abstract String getPathSuffix();
 
-    private static List<String> buildCompilerClasspath() {
-        String[] bootClasspath =
-            System.getProperty( "java.class.path" ).split( System.getProperty( "path.separator" ) );
-        String fs = System.getProperty( "file.separator" );
-        String testClasses = "target" + fs + "test-classes";
-
+    private static List<String> buildTestCompilationClasspath() {
         String[] whitelist =
             new String[] {
-                "processor" + fs + "target",  // the processor itself
-                "core" + fs + "target",  // MapStruct annotations in multi-module reactor build or IDE
-                "org" + fs + "mapstruct" + fs + "mapstruct" + fs,  // MapStruct annotations in single module build
-                "freemarker",
+                // MapStruct annotations in multi-module reactor build or IDE
+                "core" + File.separator + "target",
+                // MapStruct annotations in single module build
+                "org" + File.separator + "mapstruct" + File.separator + "mapstruct" + File.separator,
                 "guava",
                 "javax.inject",
                 "spring-beans",
                 "spring-context",
                 "joda-time" };
 
+        return filterBootClassPath( whitelist );
+    }
+
+    private static List<String> buildProcessorClasspath() {
+        String[] whitelist =
```

Comment:
Just coming to my mind, seeing this again: We may take a look at the Shrinkwrap Maven resolver API in order to build this classpath instead of copying stuff via the dependency plug-in. Nothing important, just for a dark winter night ;)